packages/dds/tree/src/simple-tree/core/treeNodeKernel.ts

```diff
@@ -127,15 +182,19 @@ export class TreeNodeKernel implements Listenable<TreeChangeEvents> {
 		return treeStatusFromAnchorCache(this.#hydrated.anchorNode);
 	}
 
-	public on<K extends keyof TreeChangeEvents>(
-		eventName: K,
-		listener: TreeChangeEvents[K],
-	): Off {
+	public on<K extends keyof KernelEvents>(eventName: K, listener: KernelEvents[K]): Off {
 		return this.#events.on(eventName, listener);
 	}
 
 	public dispose(): void {
-		this.dehydrate();
+		this.disposed = true;
```

Comment:
Would it make sense to also set `this.#hydrated = undefined` like the current code does, to avoid potential future bugs if someone forgets to check `this.disposed` before trying to use `this.#hydrated`?

Reply:
I don't like making the node look like it never hydrated when we dispose of it. My understanding is that hydration is suppose to be a one time thing and there is logic that depend on nodes never unhydrating.

If something fails to check disposed, I'd much rather have a node show up as hydrated than unhydrated if its a disposed hydrated node.

The old code was really confusing about this: I couldn't tell if it was trying to support an Unhydration actions. I don't think we want to support that, so I tried to make it clear we never do it.